src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -430,7 +444,11 @@ protected function createCrawlerFromContent($uri, $content, $type)
      */
     public function back()
     {
-        return $this->requestFromRequest($this->history->back(), false);
+        do {
+            $request = $this->history->back();
+        } while ($this->browserNavigation && in_array(spl_object_hash($request), $this->redirects));
```

Comment:
History object will throw `\LogicException('You are already on the first page.');`